cvat/apps/engine/rq.py

```diff
@@ -0,0 +1,527 @@
+
+    def __set__(self, instance: WithMeta, value: Any):
+        self.validate(value)
+        instance.meta[self._key] = value
+
+
+class ImmutableRQMetaAttribute(_GettableRQMetaAttribute):
+    pass
+
+
+class MutableRQMetaAttribute(_GettableRQMetaAttribute, _SettableRQMetaAttribute):
+    pass
+
+
+class UserRQMetaAttribute(ImmutableRQMetaAttribute):
+    def __init__(self, *, optional: bool = False, validator: Callable | None = None) -> None:
+        super().__init__(RQJobMetaField.USER, optional=optional, validator=validator)
+
+    def __get__(self, instance: WithMeta, objtype: type[WithMeta] | None = None):
+        assert RQJobMetaField.USER == self._key
+        return UserMeta(instance.meta[self._key])
+
+
+class RequestRQMetaAttribute(ImmutableRQMetaAttribute):
+    def __init__(self, *, optional: bool = False, validator: Callable | None = None) -> None:
+        super().__init__(RQJobMetaField.REQUEST, optional=optional, validator=validator)
+
+    def __get__(self, instance: WithMeta, objtype: type[WithMeta] | None = None):
+        assert RQJobMetaField.REQUEST == self._key
+        return RequestMeta(instance.meta[self._key])
+
+
+class UserMeta:
+    id: int = ImmutableRQMetaAttribute(
+        RQJobMetaField.UserField.ID, validator=lambda x: isinstance(x, int)
+    )
+    username: str = ImmutableRQMetaAttribute(
+        RQJobMetaField.UserField.USERNAME, validator=lambda x: isinstance(x, str)
+    )
+    email: str = ImmutableRQMetaAttribute(
+        RQJobMetaField.UserField.EMAIL, validator=lambda x: isinstance(x, str)
+    )
+
+    def __init__(self, meta: dict[RQJobMetaField.UserField, Any]) -> None:
+        self._meta = meta
+
+    @property
+    def meta(self) -> dict[RQJobMetaField.UserField, Any]:
+        return self._meta
+
+    def to_dict(self):
+        return self.meta
+
+
+class RequestMeta:
+    uuid = ImmutableRQMetaAttribute(
+        RQJobMetaField.RequestField.UUID, validator=lambda x: isinstance(x, str)
+    )
+    timestamp = ImmutableRQMetaAttribute(
+        RQJobMetaField.RequestField.TIMESTAMP, validator=lambda x: isinstance(x, datetime)
+    )
+
+    def __init__(self, meta: dict[RQJobMetaField.RequestField, Any]) -> None:
+        self._meta = meta
+
+    @property
+    def meta(self) -> dict[RQJobMetaField.RequestField, Any]:
+        return self._meta
+
+    def to_dict(self):
+        return self.meta
+
+
+class AbstractRQMeta(metaclass=ABCMeta):
+    def __init__(
+        self, *, job: RQJob | None = None, meta: dict[RQJobMetaField, Any] | None = None
+    ) -> None:
+        assert (job and not meta) or (meta and not job), "Only job or meta can be passed"
+        self._job = job
+        self._meta = meta
+
+    @property
+    def meta(self) -> dict[RQJobMetaField, Any]:
+        return self._job.meta if self._job else self._meta
+
+    def to_dict(self):
+        return self.meta
+
+    @classmethod
+    def for_job(cls, job: RQJob):
+        return cls(job=job)
```

Comment:
```suggestion
        return cls(job=job, meta=job.meta)
```

This would simplify the rest of the class: no need for the `if` in `def meta`, no need for the `assert` in `__init__`.